src/lib/ngsi/Metadata.cpp

```diff
@@ -193,12 +193,33 @@ std::string Metadata::render(const std::string& indent, bool comma)
 {
   std::string out     = "";
   std::string tag     = "contextMetadata";
-  std::string xValue  = stringValue;
+  std::string xValue  = toStringValue();
 
   out += startTag2(indent, tag, false, false);
   out += valueTag1(indent + "  ", "name", name, true);
   out += valueTag1(indent + "  ", "type", type, true);
-  out += valueTag1(indent + "  ", "value", xValue, false);
+
+  if (valueType == orion::ValueTypeString)
+  {
+    out += valueTag1(indent + "  ", "value", xValue, false);
+  }
+  else if (valueType == orion::ValueTypeNumber)
+  {
+    out += indent + "  " + JSON_STR("value") + ": " + xValue;
+  }
+  else if (valueType == orion::ValueTypeBoolean)
+  {
+    out += indent + "  " + JSON_STR("value") + ": " + xValue;
+  }
+  else if (valueType == orion::ValueTypeNone)
+  {
+    out += indent + "  " + JSON_STR("value") + ": " + xValue; 
+  }
+  else
+  {
+    out += indent + "  " + JSON_STR("value") + ": " + JSON_STR("unknown json type");
+  }
```

Comment:
Soon we will treat json vectors and objects also and this last `else` will never be reached. It's dead code but good to have as a security guard ...